includes/KSES.php

```diff
@@ -564,13 +564,58 @@ public function filter_kses_allowed_html( $allowed_tags ) {
 			],
 		];
 
-		$allowed_tags = array_merge( $allowed_tags, $story_components );
+		$allowed_tags = $this->array_merge_recursive_distinct( $allowed_tags, $story_components );
 
 		$allowed_tags = array_map( [ $this, 'add_global_attributes' ], $allowed_tags );
 
 		return $allowed_tags;
 	}
 
+	/**
+	 * Recursively merge a variable number of arrays, using the left array as base,
+	 * giving priority to the right array.
+	 *
+	 * Difference with native array_merge_recursive():
+	 * array_merge_recursive converts values with duplicate keys to arrays rather than
+	 * overwriting the value in the first array with the duplicate value in the second array.
+	 *
+	 * array_merge_recursive_distinct does not change the data types of the values in the arrays.
+	 * Matching keys' values in the second array overwrite those in the first array, as is the
+	 * case with array_merge.
+	 *
+	 * Freely based on information found on http://www.php.net/manual/en/function.array-merge-recursive.php
+	 *
+	 * @link https://github.com/Yoast/wordpress-seo/blob/219014897b95465fa1cd4d23a72b0c74864f6dba/inc/class-wpseo-meta.php#L938-L983
```

Comment:
Hmm... Yoast SEO is GPL, we cannot really just copy their code into ours as it's not compatible with Apache.

Reply:
This pastebin is from 2012 - https://pastebin.com/tpSqBTc2

Reply:
@swissspidy So what do I do here, do I have write the code?

Reply:
Earliest reference I can see https://www.php.net/manual/en/function.array-merge-recursive.php#89684

Reply:
IANAL, but since php.net comments and documentation is under Creative Commons it should be OK.